src/actions/WalletActions.js

```diff
@@ -67,26 +67,39 @@ export const selectWallet = (walletId: string, currencyCode: string, from?: stri
   }
 }
 
-// check if the EOS wallet is activated (via public address blank string check) and route to activation scene(s)
-export const selectEOSWallet = (walletId: string, currencyCode: string, from?: string) => (dispatch: Dispatch, getState: GetState) => {
+// check if the EOS-like (EOS, Hedera) wallet is activated (via public address blank string check) and route to activation scene(s)
+export const selectEOSLikeWallet = (walletId: string, currencyCode: string, from?: string) => (dispatch: Dispatch, getState: GetState) => {
   const state = getState()
   const currentWalletId = state.ui.wallets.selectedWalletId
   const currentWalletCurrencyCode = state.ui.wallets.selectedCurrencyCode
   const guiWallet = UI_SELECTORS.getWallet(state, walletId)
+  const coreWallet = state.core.account.currencyWallets[walletId]
   if (walletId !== currentWalletId || currencyCode !== currentWalletCurrencyCode || from === Constants.WALLET_LIST_SCENE) {
     const { publicAddress } = guiWallet.receiveAddress
     if (!publicAddress) {
       // Update all wallets' addresses. Hopefully gets the updated address for the next time
-      // We enter the EOS wallet
+      // we enter the EOS-like wallet
       dispatch(updateWalletsRequest())
     }
 
+    const activationPending = coreWallet && coreWallet.otherMethods.getIsActivationPending && coreWallet.otherMethods.getIsActivationPending()
+
     if (publicAddress) {
       // already activated
       dispatch({
         type: 'UI/WALLETS/SELECT_WALLET',
         data: { walletId, currencyCode }
       })
+    } else if (activationPending) {
+      // Hedera: wallet is pending activation, we have no course of action but to wait
+      Actions.jump(Constants.WALLET_LIST_SCENE)
```

Comment:
@paullinator 

This branch is for Hedera wallets where the activation payment was submitted and now we're just waiting on the backend to confirm the payment and create the account.

I figured if we don't want the user to really access the wallet here and so should kick them back to the wallet list scene while displaying the modal, but I couldn't figure out how to make this work. 

When this branch is triggered the modal seems to pop up over the wallet details scene for an arbitrarily different wallet instead. If I use `Actions.popTo(Constants.WalletListScene)` like I see elsewhere in the code, I get a warning that the route does not exist.

Reply:
You are seeing an arbitrary wallet details scene because you never call dispatch `UI/WALLETS/SELECT_WALLET` so the most recent selected wallet would show. I dont think you need to do this however if youre just going top hop back to the wallet list scene. Not sure if you just have a typo but the command should be 

`Actions.popTo(Constants.WALLET_LIST_SCENE)`

Not `Actions.jump(Constants.WALLET_LIST_SCENE)` and not `Actions.popTo(Constants.WalletListScene)`

Reply:
Yeah it was a typo, still having trouble with `Actions.popTo(Constants.WALLET_LIST_SCENE)`. I only tried `Actions.jump()` to see if it would make a difference.